### PR TITLE
Using React Context API for passing down the visibility prop

### PR DESCRIPTION
**Pull request by nikitatrifan**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -19,57 +18,59 @@ import { Elastic } from "./Routes/Projects/elastic";
...
-            <Guidelines visibility={visibility} />
-          </Route>
-          <Route path="/">
-            <Homepage visibility={visibility} x={position.x} y={position.y} />
-          </Route>
-        </Switch>
-        <SideLines />
-        <Grid visibility={visibility} />
-      </Analytics>
+      <GridVisibilityProvider>
```
Comment (nikitatrifan) [inline src/App.js:28]: @dnvt  every child of the provider component is going to have access to its context

File: src/Components/grid.js [hunk 2]
```diff
@@ -48,8 +49,9 @@ const useStyles = createUseStyles({
 
 // function hide() {}
 
-export function Grid({ visibility }) {
+export function Grid() {
   const classes = useStyles();
+  const visibility = useGridVisibility();
```
Comment (nikitatrifan) [inline src/Components/grid.js:54]: @dnvt it's very simple to use, like useStyles or any other hook to get access to the data

File: src/Components/utils.js [hunk 3]
```diff
@@ -168,16 +168,3 @@ export const utilsStyles = createUseStyles({
...
-
-export function Mouse() {
-  const [visibility, setVisibility] = useState(false);
-  var Mousetrap = require("mousetrap");
-
-  useEffect(() => {
-    Mousetrap.bind("ctrl+g", function () {
-      setVisibility((visibility) => !visibility);
-    });
-  });
```
Comment (nikitatrifan) [inline src/Components/utils.js:180]: when useEffect is not being used with the second argument of an array, it's going to fire the passed function every other time your app renders. Which basically was every time you were updating mouse position.

const [hidden, setHidden] = useState(false);

useEffect(() => {
  console.log('if the function is fired that means hidden state has got changed');
}, [hidden])

return <button onClick={() => setHidden(val => !val)}>toggle</button>

File: src/Components/grid.js (hunk 2 above)
Comment (dnvt) [reply src/Components/grid.js:54]: That's so cool! Thanks @nikitatrifan 
So now I could use the same for the mouse position x and y used for Fontimation component, which is in 2 Routes 👍👍

File: src/Components/utils.js (hunk 3 above)
Comment (dnvt) [reply src/Components/utils.js:180]: Ohoh... :)

**Pull request merged**